### ISSUE TITLE
GLES3 support on Android

### ISSUE THREAD

**Issue by loganmc10** [libretro/RetroArch#3345]
I'm working on a libretro core that will have GLES3 support. It looks like RetroArch for Android only supports GLES2 at the moment.

Also, compiling RetroArch on my laptop (Intel MESA, GLES3 support), with ```./configure --enable-gles3``` doesn't work either, when I launch the core I get this:

```
RetroArch [ERROR] :: Requesting OpenGLES3 context, but RetroArch is compiled against OpenGL. Cannot use HW context.
```

**Comment by loganmc10** [comment]
I've been trying to do some reading on how to get an Android App to dynamically support GLES2 or GLES3, but I haven't really found a way to do it, is there someone with some Android experience that might be able to point me in the right direction on this?

**Comment by loganmc10** [comment]
These are the messages I get from the Android App (using the 08-22 nightly)

```
Environ SET_HW_RENDER.
Requesting unknown context.
```

And then a little later on:
```
Android EGL: GLES version = 2.
```

And then even later:
```
[GL]: Version: OpenGL ES 3.1 v1.r7p0-02rel0.a6c0939b3050616905647b55b149c24b.
```

**Comment by loganmc10** [comment]
Just looking at the code, it seems that defining HAVE_OPENGLES3 during the build of the app should pretty much solve the problem, the only issue I can see if that it will trigger this code:

**Comment by loganmc10** [comment]
This should be fixed by 8b34a009dc06b8d20c36e60fbdb4f437272feb36, but I'll try the next nightly that comes out to confirm, then I'll close

**Comment by loganmc10** [comment]
Fixed now
